cmd/river/go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/riverqueue/river/cmd/river
 
 go 1.21.4
 
+toolchain go1.21.6
```

Comment:
Can we take `toolchain` out? Go should be able to pick it up from the version above.